core/jobs/batch_jobs/question_migration_jobs.py

```diff
@@ -0,0 +1,286 @@
+            question_summary.version = question_version
+            question_summary.validate()
+        except Exception as e:
+            logging.exception(e)
+            return result.Err((question_summary.id, e))
+
+        question_summary_model = question_models.QuestionSummaryModel(
+            id=question_summary.id,
+            question_model_last_updated=question_summary.last_updated,
+            question_model_created_on=question_summary.created_on,
+            question_content=question_summary.question_content,
+            misconception_ids=question_summary.misconception_ids,
+            interaction_id=question_summary.interaction_id,
+            version=question_summary.version
+        )
+        question_summary_model.update_timestamps()
+        return result.Ok((question_summary.id, question_summary_model))
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns a PCollection of results from the question migration.
+
+        Returns:
+            PCollection. A PCollection of results from the
+            question_id_to_version migration.
+        """
+        all_question_versions = (
+            self.pipeline
+            | 'Get all non-deleted question models' >> (
+                ndb_io.GetModels(question_models.QuestionModel.get_all()))
+            # Pylint disable is needed becasue pylint is not able to correclty
+            # detect that the value is passed through the pipe.
+            | 'Add question keys and extract version' >> beam.Map( # pylint: disable=no-value-for-parameter
+                lambda model: (model.id, model.version)
+            )
+        )
+
+        question_summary_models = (
+            self.pipeline
+            | 'Get all non-deleted question summary models' >> (
+                ndb_io.GetModels(
+                    question_models.QuestionSummaryModel.get_all())
+                )
+            # Pylint disable is needed because pylint is not able to correctly
+            # detect that the value is passed through the pipe.
+            | 'Add question summary keys' >> beam.WithKeys( # pylint: disable=no-value-for-parameter
+                lambda question_summary_model: question_summary_model.id)
+        )
+
+        question_objects = (
+            {
+                'version': all_question_versions,
+                'question_summary_model': question_summary_models,
+            }
+            | 'Merge objects' >> beam.CoGroupByKey()
+            | 'Get rid of ID' >> beam.Values() # pylint: disable=no-value-for-parameter
+            | 'Reorganize the objects' >> beam.Map(lambda objects: {
+                    'version': objects['version'][0],
+                    'question_summary_model': objects[
+                        'question_summary_model'][0],
+                })
+        )
+
+        all_updated_question_summary_results = (
+            question_objects
+            | 'Update question summary models' >> beam.Map(
```

Comment:
```suggestion
            | 'Update question summary models' >> beam.Map(
```

Optional* I'm not sure whether there's any benefit of having separate pipeline/collections here? If not, should we combine it? (It will also help us rename `question_objects` (no question object involved, right?) to `updated_question_summary_results`.)

Reply:
Done